spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshIT.java

```diff
@@ -109,13 +115,68 @@ void testActuatorRefresh() {
 
 		WireMock.verify(WireMock.postRequestedFor(WireMock.urlEqualTo("/actuator/refresh")));
 		util.deleteAndWait(NAMESPACE, configMap, null);
+
+		configWatcher(Phase.DELETE, false);
 	}
 
-	private void configWatcher(Phase phase) {
+	/*
+	 * same test as above, but reload is disabled.
+	 */
+	@Test
+	void testActuatorRefreshReloadDisabled() {
+		configWatcher(Phase.CREATE, true);
+
+		WireMock.configureFor(WIREMOCK_HOST, WIREMOCK_PORT, WIREMOCK_PATH);
+		await().timeout(Duration.ofSeconds(60))
+				.until(() -> WireMock
+						.stubFor(WireMock.post(WireMock.urlEqualTo("/actuator/refresh"))
+								.willReturn(WireMock.aResponse().withBody("{}").withStatus(200)))
+						.getResponse().wasConfigured());
+
+		// Create new configmap to trigger controller to signal app to refresh
+		V1ConfigMap configMap = new V1ConfigMapBuilder().editOrNewMetadata().withName("service-wiremock")
+				.addToLabels("spring.cloud.kubernetes.config", "true").endMetadata().addToData("foo", "bar").build();
+		util.createAndWait(NAMESPACE, configMap, null);
+
+		// Wait a bit before we verify
+		await().atMost(Duration.ofSeconds(30)).until(
+				() -> !WireMock.findAll(WireMock.postRequestedFor(WireMock.urlEqualTo("/actuator/refresh"))).isEmpty());
+
+		Assertions.assertTrue(logs().contains("creating NOOP strategy because reload is disabled"));
+		// nothing related to 'ConfigReloadUtil' is present in logs
+		// this proves that once we disable reload everything still works
+		Assertions.assertFalse(logs().contains("ConfigReloadUtil"));
+
+		WireMock.verify(WireMock.postRequestedFor(WireMock.urlEqualTo("/actuator/refresh")));
+		util.deleteAndWait(NAMESPACE, configMap, null);
+
+		configWatcher(Phase.DELETE, true);
+	}
+
+	private void configWatcher(Phase phase, boolean disableReload) {
 		V1ConfigMap configMap = (V1ConfigMap) util
 				.yaml("config-watcher/spring-cloud-kubernetes-configuration-watcher-configmap.yaml");
 		V1Deployment deployment = (V1Deployment) util
 				.yaml("config-watcher/spring-cloud-kubernetes-configuration-watcher-http-deployment.yaml");
+
+		List<V1EnvVar> envVars = new ArrayList<>(
+				Optional.ofNullable(deployment.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv())
+						.orElse(new ArrayList<>()));
+
+		V1EnvVar commonsDebug = new V1EnvVar()
+				.name("LOGGING_LEVEL_ORG_SPRINGFRAMEWORK_CLOUD_KUBERNETES_COMMONS_CONFIG_RELOAD").value("DEBUG");
+		V1EnvVar watcherDebug = new V1EnvVar()
+				.name("LOGGING_LEVEL_ORG_SPRINGFRAMEWORK_CLOUD_KUBERNETES_CONFIGURATION_WATCHER").value("DEBUG");
+
+		envVars.add(commonsDebug);
+		envVars.add(watcherDebug);
+
+		if (disableReload) {
+			V1EnvVar disableReloadEnvVar = new V1EnvVar().name("SPRING_CLOUD_KUBERNETES_RELOAD_ENABLED").value("FALSE");
```

Comment:
disable reload: `SPRING_CLOUD_KUBERNETES_RELOAD_ENABLED` - app still works. This test fails, without the fix that we did here. 

I've also "tested" a build with disabling reload everywhere - that is, in the configuration watcher `application.yaml` and it passed all the integration tests.